zipline/pipeline/factors/factor.py

```diff
@@ -833,6 +834,90 @@ def linear_regression(self, target, regression_length, mask=NotSpecified):
             mask=mask,
         )
 
+    @float64_only
+    def winsorize(self,
+                  limits,
+                  inclusive=(True, True),
+                  mask=NotSpecified,
+                  groupby=NotSpecified):
+        """
+        Construct a Factor returns a winsorized row for results. Winsorizing
+        clips the input values to fixed percentiles. The (limits[0])th lowest
+        values are set to the value at the (limits[0])th percentile. The values
+        above the (limits[1])th percentiles are set to the value at the
+        (limits[1])th percentile. This is useful when limiting the impact of
+        extreme values.
+
+        If ``mask`` is supplied, ignore values where ``mask`` returns False
+        when computing row means and standard deviations, and output NaN
+        anywhere the mask is False.
+
+        If ``groupby`` is supplied, compute by partitioning each row based on
+        the values produced by ``groupby``, winsorizing the partitioned arrays,
+        and stitching the sub-results back together.
+
+        Parameters
+        ----------
+        limits : None, tuple of float, optional
```

Comment:
Reading through the examples from the tests, I'm still not thrilled with this API for limits.  Reading `f.winsorize(limits=33)` in the tests, for example, feels awkward to me.  It's not clear in that usage that `limits` is a percentile and not an absolute value (an issue that would persist even for more reasonable limits like `5`).

I spent a while thinking about this without coming up with any suggestions that I really liked.  Let's talk about this first thing tomorrow and figure out a plan so that we can get this merged.